cirq-core/cirq/transformers/transformer_primitives.py

```diff
@@ -281,6 +281,73 @@ def apply_merge_func(op1: ops.Operation, op2: ops.Operation) -> Optional[ops.Ope
     return _to_target_circuit_type(ret_circuit, circuit)
 
 
+def merge_operations_to_circuit_op(
+    circuit: CIRCUIT_TYPE,
+    can_merge: Callable[[Sequence['cirq.Operation'], Sequence['cirq.Operation']], bool],
+    *,
+    tags_to_ignore: Sequence[Hashable] = (),
+    merged_circuit_op_tag: str = "Merged connected component",
+) -> CIRCUIT_TYPE:
+    """Merges connected components of operations and wraps each component into a circuit operation.
+
+    Uses `cirq.merge_operations` to identify connected components of operations.
```

Comment:
Do we want to note here that like `merge_operations` this WILL affect the moment structure of the produced circuit operation ? This makes me wonder at a high level, should we have something to "circuit_op"ify something that preservers moment structure and then just be able to call merge_operations on the resulting CircuitOp ?